### PR TITLE
adding views for auth

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
-# Create your views here.
+from users.models import User
+from users.serializers import UserPublicSerializer
+
+
+class GetUser(APIView):
```
Comment (revimi) [inline users/views.py]: Why do we need this view?

File: users/views.py [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+    def get(self, request):
+        users_objects = User.objects.all()
+        serializer = UserPublicSerializer(users_objects, many=True)
+        return Response({"status": 200, "payload": serializer.data})
+
+
+class RegisterUser(APIView):
+    def post(self, request):
+        serializer = UserPublicSerializer(data=request.data)
```
Comment (revimi) [inline users/views.py]: How does the user set their password?

File: users/views.py (hunk 1 above)
Comment (marinaRomanchuk) [reply users/views.py]: Removed

File: users/views.py (hunk 2 above)
Comment (marinaRomanchuk) [reply users/views.py]: Serializer and new view for sign up were implemented

File: users/serializers.py [hunk 3]
```diff
@@ -25,3 +26,24 @@ class Meta(UserPublicSerializer.Meta):
             "date_of_birth",
             "profile_photo",
         ]
+
+
+class SetupSerializer(serializers.Serializer):
```
Comment (revimi) [inline users/serializers.py]:
```suggestion
class SignUpSerializer(serializers.Serializer):
```

**Review by revimi — CHANGES_REQUESTED** [review]

File: users/serializers.py (hunk 3 above)
Comment (marinaRomanchuk) [reply users/serializers.py]: Done

File: instagram_own_version/urls.py [hunk 4]
```diff
@@ -13,9 +13,13 @@
...
+from django.conf.urls import url
 from django.contrib import admin
-from django.urls import path
+from django.urls import include, path
+from rest_framework.authtoken import views as token_views
 
 urlpatterns = [
-    path('admin/', admin.site.urls),
+    path("admin/", admin.site.urls),
+    path("", include("users.urls")),
```
Comment (revimi) [inline instagram_own_version/urls.py]:
```suggestion
    path("api/", include("users.urls")),
```

File: users/serializers.py [hunk 5]
```diff
@@ -25,3 +26,24 @@ class Meta(UserPublicSerializer.Meta):
...
+    password = serializers.CharField(
+        write_only=True, required=True, validators=[validate_password]
+    )
+
+    class Meta:
+        model = User
+        fields = ("username", "password")
+
+    def create(self, validated_data):
+        user = User.objects.create(
```
Comment (revimi) [inline users/serializers.py:49]: We should check if the user already exists. We can do it in the `validate` method

File: users/views.py [hunk 6]
```diff
@@ -1,3 +1,11 @@
...
+from rest_framework import generics
+from rest_framework.permissions import AllowAny
 
-# Create your views here.
+from users.models import User
+from users.serializers import SignupSerializer
+
+
+class SignupView(generics.CreateAPIView):
+    queryset = User.objects.all()
```
Comment (revimi) [inline users/views.py]: Looks like this line is excess. This view does not use it as it does not have retrieve or list endpoints

**Review by revimi — CHANGES_REQUESTED** [review]
:+1:

File: instagram_own_version/urls.py (hunk 4 above)
Comment (marinaRomanchuk) [reply instagram_own_version/urls.py]: Done

File: users/serializers.py (hunk 5 above)
Comment (marinaRomanchuk) [reply users/serializers.py:49]: Validate method was implemented

File: users/views.py (hunk 6 above)
Comment (marinaRomanchuk) [reply users/views.py]: Removed

**Review by revimi — APPROVED** [review]
Good work!

**Pull request merged**